src/basic/fs-util.c

```diff
@@ -288,8 +288,21 @@ int fchmod_umask(int fd, mode_t m) {
 
 int fchmod_opath(int fd, mode_t m) {
         /* This function operates also on fd that might have been opened with
-         * O_PATH. Indeed fchmodat() doesn't have the AT_EMPTY_PATH flag like
-         * fchownat() does. */
+         * O_PATH. The tool set we have is non-intuitive:
+         * - fchmod(2) only operates on open files (i. e., fds with an open file description);
+         * - fchmodat(2) does not have a flag arg like fchownat(2) does, so no way to pass AT_EMPTY_PATH;
+         *   + it should not be confused with the libc fchmodat(3) interface, which adds 4th flag argument,
+         *     but does not support AT_EMPTY_PATH (only supports AT_SYMLINK_NOFOLLOW);
+         * - fchmodat2(2) supports all the AT_* flags, but is still very recent.
+         *
+         * We try to use fchmodat2(), and, if it is not supported, resort
+         * to the /proc/self/fd dance. */
+
+        if (fchmodat2(fd, "", m, AT_EMPTY_PATH) < 0) {
+                if (errno != ENOSYS)
+                        return -errno;
+        } else
+                return 0;
```

Comment:
How about the following:

```suggestion
        if (fchmodat2(fd, "", m, AT_EMPTY_PATH) >= 0)
                return 0;
        if (errno != ENOSYS)
                return -errno;
```

Reply:
Not sure about that suggestion to invert the check.
When we need to add a check for an error, we generally use `< 0`.

Reply:
We adopt such practice in several other places too. And IMO this makes logic clearer.